src/opm/output/eclipse/AggregateActionxData.cpp

```diff
@@ -0,0 +1,587 @@
+            };
+        }
+
+    template <class ZACTArray>
+    void staticContrib(const Opm::Action::ActionX& actx, ZACTArray& zAct)
+    {
+        // entry 1 is udq keyword
+        zAct[0] = actx.name();
+    }
+    } // zAct
+
+    namespace zLACT {
+
+         Opm::RestartIO::Helpers::WindowedArray<
+            Opm::EclIO::PaddedOutputString<8>
+        >
+        allocate(const std::vector<int>& actDims)
+        {
+            using WV = Opm::RestartIO::Helpers::WindowedArray<
+                Opm::EclIO::PaddedOutputString<8>
+            >;
+
+            return WV {
+                WV::NumWindows{ static_cast<std::size_t>(actDims[0]) },
+                WV::WindowSize{ static_cast<std::size_t>(actDims[4]) }
+            };
+        }
+        
+    template <class ZLACTArray>
+    void staticContrib(const Opm::Action::ActionX& actx, int noEPrZlact, ZLACTArray& zLact)
+        {
+            std::size_t ind = 0;
+            int l_sstr = 8;
+            int max_l_str = 128;
+            // write out the schedule input lines
+            const auto& schedule_data = actx.keyword_strings();
+            for (auto z_data : schedule_data) {
+                int n_sstr =  z_data.size()/l_sstr;
+                if (static_cast<int>(z_data.size()) > max_l_str) {
+                    std::cout << "Too long input data string (max 128 characters): " << z_data << std::endl;
+                    throw std::invalid_argument("Actionx: " + actx.name());
+                }
+                else {
+                    for (int i = 0; i < n_sstr; i++) {
+                        zLact[ind + i] = z_data.substr(i*l_sstr, l_sstr);
+                    }
+                    //add remainder of last non-zero string
+                    if ((z_data.size() % l_sstr) > 0)
+                        zLact[ind + n_sstr] = z_data.substr(n_sstr*l_sstr);
+                }
+                ind += static_cast<std::size_t>(noEPrZlact);
+            }
+        }      
+    } // zLact
+    
+    namespace zACN {
+
+         Opm::RestartIO::Helpers::WindowedArray<
+            Opm::EclIO::PaddedOutputString<8>
+        >
+        allocate(const std::vector<int>& actDims)
+        {
+            using WV = Opm::RestartIO::Helpers::WindowedArray<
+                Opm::EclIO::PaddedOutputString<8>
+            >;
+
+            return WV {
+                WV::NumWindows{ static_cast<std::size_t>(actDims[0]) },
+                WV::WindowSize{ static_cast<std::size_t>(actDims[5]) }
+            };
+        }
+  
+    template <class ZACNArray>
+    void staticContrib(const Opm::Action::ActionX& actx, ZACNArray& zAcn)
+        {
+            std::size_t ind = 0;
+            int noEPZacn = 13;
+            // write out the schedule Actionx conditions
+            const auto& actx_cond = actx.conditions();
+            for (auto z_data : actx_cond) {
+                // left hand quantity
+                if ((z_data.lhs.quantity.substr(0,1) != "D") &&
```

Comment:
Instead of creating the substring object you could just do: `z_data.lhs.quantity[0] != 'D'`